### ISSUE TITLE
bulk stock adding, search box focus, custom location field and stock reports

### ISSUE THREAD

**Issue by dmd-dmz** [snipe/snipe-it#215]
SnipeItApp looks really awesome after playing around with the demo, really impressed with the project! Id like to make a few suggestions that im hoping would benefit people in my position and the community: The main thing is the ability to "bulk add" assets into the system. I Frequently purchase 50-100 units at a time which last a week before the next batch came in.

1) If we could bulk add similar items but differentiate with serial numbers (auto incrementing Asset IDs too), that would be fantastic.

2) Focus on search box ready for barcode scanner input would be nice too.

3) Custom location field. Instead of "checked out to" a user, we can create a custom checked out location. Saves assigning to a user account everytime & useful for rapid deployments where items go to a site and not an individual.

4) Stock management & reporting pages. Minimum reorder levels etc. 

Keep up the good work on this awesome project :)
dmd-dmz

**Comment by snipe** [comment]
There is a rough cli bulk importer on `develop` now. 

Usage:
`php artisan import:csv app/private_uploads/yourcsv.csv --domain=yourdomain.com`

It will be looking for a CSV in the format of:

`Assigned To, Category, Asset Model Name, Manufacturer, Model Number,Serial Number, Asset Tag, Location`

It will split the user’s name, creating firstname and lastname, will generate an email address using your domain and the FirstInitialLastName@yourdomain.com pattern, will generate a password for them, and will create the asset models, locations, category, etc if they don't currently exist.

It currently does not do any validation other than checking to see if the record already exists in the database.

It determines if the user exists by looking up the email address (which has been generated as FirstInitialLastName@yourdomain.com, and may not be correct for all email naming conventions.)

If you want to test this, **BACK UP YOUR DATABASE FIRST**. This is experimental, and it might shit all over your data. I have successfully used this cli tool to import a CSV of over 2,000 assets, but there's some data cleansing and checking that needs to be built in before it's bulletproof.

Closing this now, since there are multiple issues reported in it, and that's too difficult to track. Pretty sure there are issues open for everything else anyway though. (If not, @dmd-dmz, feel free to open separate tickets for the remaining issues.)
